From 2d20a354c76dd21dfc7d85d4f39020609e7fcf16 Mon Sep 17 00:00:00 2001
From: Lei Qian <lei.qian@amlogic.com>
Date: Wed, 27 Mar 2019 16:05:29 +0800
Subject: [PATCH] audio: fix sinc resampler leads audioserver excessive memory
 consumption [1/1]

PD# TV-3794

Problem:
sinc resampler leads audioserver excessive memory consumption

Solution:
after convert free memory

Verify:
verify by x301

Change-Id: I73f04569cf496e20d1eb0bdefa1afbd0a4ec2a66
Signed-off-by: Lei Qian <lei.qian@amlogic.com>
---
 media/libaudioprocessing/RecordBufferConverter.cpp | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/media/libaudioprocessing/RecordBufferConverter.cpp b/media/libaudioprocessing/RecordBufferConverter.cpp
index e559c34..65ed195 100644
--- a/media/libaudioprocessing/RecordBufferConverter.cpp
+++ b/media/libaudioprocessing/RecordBufferConverter.cpp
@@ -157,6 +157,9 @@ size_t RecordBufferConverter::convert(void *dst,
 
          memcpy_by_audio_format(dst, mDstFormat, convert, AUDIO_FORMAT_PCM_16_BIT,
                  frames * mDstChannelCount);
+
+         if (convert)
+             free(convert);
 #else //original code
          // reallocate buffer if needed
          if (mBufFrameSize != 0 && mBufFrames < frames) {
-- 
1.9.1

